services/processor/native/call_test.go

```diff
@@ -0,0 +1,105 @@
+package native
+
+import (
+	"github.com/stretchr/testify/require"
+	"reflect"
+	"testing"
+)
+
+func TestPrepareMethodArgumentsForCallWithUint32(t *testing.T) {
+	s := &service{}
+
+	methodInstance := func(a uint32) uint32 {
+		return a
+	}
+
+	args := argsToArgumentArray(uint32(1997))
+
+	inValues, err := s.prepareMethodInputArgsForCall(methodInstance, args, "funcName")
+	require.NoError(t, err)
+
+	outValues := reflect.ValueOf(methodInstance).Call(inValues)
+	require.EqualValues(t, 1997, outValues[0].Uint())
+}
+
+func TestPrepareMethodArgumentsForCallWithByteArray(t *testing.T) {
+	s := &service{}
+
+	methodInstance := func(a []byte) []byte {
+		return a
+	}
+
+	args := argsToArgumentArray([]byte("hello"))
+
+	inValues, err := s.prepareMethodInputArgsForCall(methodInstance, args, "funcName")
+	require.NoError(t, err)
+
+	outValues := reflect.ValueOf(methodInstance).Call(inValues)
+	require.EqualValues(t, []byte("hello"), outValues[0].Bytes())
+}
+
+func TestPrepareMethodArgumentsForCallWithArrayOfVariableLength(t *testing.T) {
+	s := &service{}
+
+	methodInstance := func(a ...string) []string {
+		return a
+	}
+
+	args := argsToArgumentArray("one", "two")
+
+	inValues, err := s.prepareMethodInputArgsForCall(methodInstance, args, "funcName")
+	require.NoError(t, err)
+
+	outValues := reflect.ValueOf(methodInstance).Call(inValues)
+	require.EqualValues(t, []string{"one", "two"}, outValues[0].Interface().([]string))
+}
+
+func TestPrepareMethodArgumentsForCallWithArrayOfByteArrays(t *testing.T) {
+	s := &service{}
+
+	methodInstance := func(a ...[]byte) [][]byte {
+		return a
+	}
+
+	args := argsToArgumentArray([]byte("one"), []byte("two"))
+
+	inValues, err := s.prepareMethodInputArgsForCall(methodInstance, args, "funcName")
+	require.NoError(t, err)
+
+	outValues := reflect.ValueOf(methodInstance).Call(inValues)
+	require.EqualValues(t, [][]byte{[]byte("one"), []byte("two")}, outValues[0].Interface())
+}
+
+func TestPrepareMethodArgumentsForCallWithTwoByteArrays(t *testing.T) {
+	s := &service{}
+
+	methodInstance := func(a []byte, b []byte) [][]byte {
+		return [][]byte{a, b}
+	}
+
+	args := argsToArgumentArray([]byte("one"), []byte("two"))
+
+	inValues, err := s.prepareMethodInputArgsForCall(methodInstance, args, "funcName")
+	require.NoError(t, err)
+
+	outValues := reflect.ValueOf(methodInstance).Call(inValues)
+	require.EqualValues(t, [][]byte{[]byte("one"), []byte("two")}, outValues[0].Interface())
+}
+
+func TestPrepareMethodArgumentsForCallWithIncorrectNumberOfArgs(t *testing.T) {
```

Comment:
this one is "too many arguments". Do we need one for "too few"?

Reply:
It is covered by other, more high level tests here: https://github.com/orbs-network/orbs-network-go/blob/656c5b03ada55c68ab51d5f0b34d737184f7448f/services/processor/native/test/call_argument_test.go#L41